src/Avalonia.X11/X11Window.cs

```diff
@@ -391,9 +391,6 @@ void OnEventSync(XEvent ev)
 
                         Dispatcher.UIThread.RunJobs(DispatcherPriority.Layout);
                     }, DispatcherPriority.Layout);
-                if (_useRenderWindow)
```

Comment:
@kekekeks I assume this block was necessary for some functionality, however it was causing the window to get resized to an old requested size on my system, e.g.:

- `ArrangeOverride` requests size A
- `ArrangeOverride` requests size B
- This code is called with size A and undoes the size B

Note I'm running Ubuntu with wayland and my system has two monitors attached with different scalings. I don't know if that is relevant.

Reply:
If `_useRenderWindow` is true, an separate *child* window is created for rendering. It has to be resized to match the size of the main window, otherwise rendering would break

Reply:
I'm not sure why it undoes the size B, since it shouldn't affect the main window. Please, check if `event` member (https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.X11/X11Structs.cs#L320) has its value set to `_handle` or `_renderHandle`. We might want to check for `event` instead of `window` here https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.X11/X11Window.cs#L354